File: src/main/java/com/todoist/pojo/TodoistObject.java
package com.todoist.pojo;

class TodoistObject {
    private long id;
    private boolean deleted;

    public TodoistObject(long id, boolean deleted) {
        this.id = id;
        this.deleted = deleted;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public boolean hasTempId() {
        return id < 0;
    }

    public boolean isDeleted() {
        return deleted;
    }

    public void setDeleted(boolean deleted) {
        this.deleted = deleted;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) {
            return true;
        }
        if (other == null || getClass() != other.getClass()) {
            return false;
        }

        return id == ((TodoistObject) other).id;
    }

    @Override
    public int hashCode() {
        return (int) (id ^ (id >>> 32));
    }
}

File: src/main/java/com/todoist/pojo/TodoistObject.kt
package com.todoist.pojo

open class TodoistObject(open var id: Long, open var isDeleted: Boolean) {
    open fun hasTempId() = id < 0

    override fun equals(other: Any?) = when {
        this === other -> true
        other == null || javaClass != other.javaClass -> false
        else -> id == (other as TodoistObject).id
    }

    override fun hashCode() = (id xor id.ushr(32)).toInt()
}
